### PR TITLE
Playwright e2e

### PR DESCRIPTION
**Pull request by DukeCephalopod**
* Move all our client-only pages (all of them) from `app/` to `pages/` (with a few tweaks). I'm not necessarily planning to stay on next.js long term, but the immediate result is that "setting page title actually works", which this changeset includes, which I only found out because:
* Set up playwright with some very barebones starter tests to assert that each page's `<title>` is correct. Which they weren't, because they were in `app/`, not `pages/`, which get parsed & loaded differently :upside_down_face: 
* Remove `@mui/material`, which was only being used in the Guide (replaced with normal components) and fucking up the root layout (now unfucked)
* Move pages/ and components/ into src/ so our poor root dir is slightly less cluttered

### CODE REVIEW

File: .github/workflows/playwright.yml
Comment (Fyko) [inline .github/workflows/playwright.yml:1]: use the existing node setup and yarn install steps from the cd and ci steps

File: package.json [hunk 1]
```diff
@@ -6,46 +6,47 @@
 		"dev": "next dev",
 		"build": "next build",
 		"start": "next start",
-		"lint": "prettier --check . && eslint_d --format=pretty \"app/**\" \"components/**\"",
-		"format": "prettier --write . && eslint_d --fix --format=pretty \"app/**\" \"components/**\"",
-		"test": "mocha {app,components}/**/*.spec.ts",
-		"typecheck": "tsc --noEmit"
+		"lint": "prettier --check . && eslint_d --format=pretty src",
```
Comment (Fyko) [inline package.json:9]: what is eslint_d

File: playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { defineConfig, devices } from '@playwright/test';
+
+// purposefully not using ??: we want to replace empty empty string with default
+const BASE_URL = process.env.BASE_URL || 'http://localhost:3000';
+
+let addlConfig = {};
+if (BASE_URL.includes('localhost') || BASE_URL.includes('127.0.0.1')) {
```
Comment (Fyko) [inline playwright.config.ts:7]: Array#includes

**Review by Fyko — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (DukeCephalopod) [reply package.json:9]: daemonized eslint, faster
```
╔═(robert@erebus)════(~/code/majeggstics)
╚══~> time yarn eslint

________________________________________________________
Executed in    2.18 secs    fish           external
   usr time    2.70 secs  763.00 micros    2.70 secs
   sys time    0.28 secs  122.00 micros    0.28 secs

╔═(robert@erebus)════(~/code/majeggstics)
╚══~> time yarn eslint_d

________________________________________________________
Executed in    2.15 secs      fish           external
   usr time  797.49 millis  717.00 micros  796.77 millis
   sys time   83.95 millis  115.00 micros   83.84 millis
```

File: playwright.config.ts (hunk 2 above)
Comment (DukeCephalopod) [reply playwright.config.ts:7]: i don't know what this comment means? that's [String#includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) so idk where i could use an array here

unless you want
```typescript
if (['localhost', '127.0.0.1'].some((host: string) => BASE_URL.includes(host)) {
```

but i don't really like that for just two elements :thinking:

File: .github/workflows/playwright.yml
Comment (DukeCephalopod) [reply .github/workflows/playwright.yml:1]: :heavy_check_mark:

File: playwright.config.ts (hunk 2 above)
Comment (Fyko) [reply playwright.config.ts:7]: that also words and is more dry

Comment (DukeCephalopod) [reply playwright.config.ts:7]: i think i'm still leaning towards two-includes over array-some --- if i thought there was a chance we'd ever add a third thing i'd definitely do array-some, but i really don't think there is. two-includes fits all on one line neatly so there's not really a big maintenance/readability benefit right now either, so it kind of feels like the worse performance (small tho it is) doesn't get us much

**Review by Fyko — APPROVED** [review]

**Pull request merged**